corehq/apps/linked_domain/remote_accessors.py

```diff
@@ -57,22 +57,24 @@ def _convert_app_from_remote_linking_source(app_json):
     return app
 
 
-def pull_missing_multimedia_for_app(app):
-    missing_media = _get_missing_multimedia(app)
+def pull_missing_multimedia_for_app(app, old_multimedia_ids=None):
+    missing_media = _get_missing_multimedia(app, old_multimedia_ids)
     remote_details = app.domain_link.remote_details
     _fetch_remote_media(app.domain, missing_media, remote_details)
     if toggles.CAUTIOUS_MULTIMEDIA.enabled(app.domain):
-        still_missing_media = _get_missing_multimedia(app)
+        still_missing_media = _get_missing_multimedia(app, old_multimedia_ids)
         if still_missing_media:
             raise MultimediaMissingError(_(
                 'Application has missing multimedia even after an attempt to re-pull them. '
                 'Please try re-pulling the app. If this persists, report an issue.'
             ))
 
 
-def _get_missing_multimedia(app):
+def _get_missing_multimedia(app, old_multimedia_ids=None):
     missing = []
     for path, media_info in app.multimedia_map.items():
+        if old_multimedia_ids and media_info['multimedia_id'] in old_multimedia_ids:
+            continue
```

Comment:
I don't follow this change, how do we know it's not missing just because it's in the current app's mm map?

Reply:
if it's already present in the last multimedia map(before the current pull) i.e the mm was already pulled in the last master pull, its assumed to be present.

Reply:
I'm not sure that assumption holds. Since MM is pulled independently of the app is it not possible for the MM to be present in the app doc but not in the DB?

Reply:
@snopoke Do you mean if the app is pulled, doc is updated but the process fails in between when pulling multimedia? Working on that possibility in https://dimagi-dev.atlassian.net/browse/ICDS-731

Though we would want to avoid that case but If that happens and the process does fail leaving some MM missing, the user can use the "Sync multimedia" feature to get things aligned?

Reply:
But with your changes won't there be the chance that you won't even check if the MM exists since it will already be in app MM map?

Reply:
@mkangia I'm still sure that this is correct.

Reply:
offlined with @snopoke , there was a misunderstanding that we would not check all multimedia in case of a multimedia sync initiated by user.